controllers/admin/self-managed/UpdatePageVersionChoiceController.php

```diff
@@ -206,17 +207,40 @@ private function getRequirements(): array
     public function save(): JsonResponse
     {
         $channel = $this->request->get(self::FORM_FIELDS['channel']);
+        $isLocal = $channel === self::FORM_OPTIONS['local_value'];
 
-        $controller = new UpdateConfig($this->upgradeContainer);
-        $controller->init();
-        $controller->run();
+        $requestConfig = $this->request->request->all();
+
+        $this->upgradeContainer->initPrestaShopCore();
+
+        $error = $this->upgradeContainer->getConfigurationValidator()->validate($requestConfig);
+
+        if ($isLocal && empty($error)) {
+            $error = $this->upgradeContainer->getLocalChannelConfigurationValidator()->validate($requestConfig);
+        }
+
+        if (empty($error)) {
+            if ($isLocal) {
+                $file = $requestConfig['archive_zip'];
+                $fullFilePath = $this->upgradeContainer->getProperty(UpgradeContainer::DOWNLOAD_PATH) . DIRECTORY_SEPARATOR . $file;
+                $config['archive_version_num'] = (new PrestashopVersionService($this->upgradeContainer->getZipAction()))->extractPrestashopVersionFromZip($fullFilePath);
+            }
+
+            $config = $this->upgradeContainer->getUpgradeConfiguration();
```

Comment:
Config is reassigned afterwards? the `if ($isLocal)` is therefore not useful, right?